packages/core/src/components/divider/_divider.scss

```
@@ -15,4 +15,8 @@ $divider-margin: $pt-grid-size * 0.5 !default;
   .#{$ns}-dark & {
     border-color: $pt-dark-divider-white;
   }
+
+  &-compact {
```

I'd suggest we name this class off of the top level `bp*` namespace rather than nesting it under `divider`.
```suggestion
  .#{$ns}-compact {
```
This would be more in line with other components that have this "compact" prop/class like Callout and Card:
https://github.com/palantir/blueprint/blob/1782045c7837b73b67cf743241214f59f46c6015/packages/core/src/components/callout/_callout.scss#L62-L73
https://github.com/palantir/blueprint/blob/1782045c7837b73b67cf743241214f59f46c6015/packages/core/src/components/card/_card.scss#L43-L45